### PR TITLE
Retry creating object in listener on slug colision

### PR DESCRIPTION
**Pull request by gregorjerse**
EXP-521

### CODE REVIEW

File: docs/CHANGELOG.rst [hunk 1]
```diff
@@ -17,6 +17,11 @@ Added
 - Add notify_create method on Subscription model in observers to enable
   sending notification when object is created
 
+Changed
+-------
+- When slug colision occurs in the listener when creating new objects retry up
```
Comment (romunov) [inline docs/CHANGELOG.rst:22]: Correct spelling: collision

File: resolwe/flow/models/base.py [hunk 2]
```diff
@@ -14,6 +14,10 @@
 MAX_SLUG_RETRIES = 10
 
 
+class UniqueSlugError(IntegrityError):
+    """The error indicates slug colision."""
```
Comment (romunov) [inline resolwe/flow/models/base.py:18]: Correct spelling: collision

File: resolwe/flow/models/base.py [hunk 3]
```diff
@@ -106,6 +110,6 @@ def save(self, *args, **kwargs):
 
                 raise
         else:
-            raise IntegrityError(
+            raise UniqueSlugError(
```
Comment (romunov) [inline resolwe/flow/models/base.py:113]: Perhaps error could print what the max number of retries is?

File: resolwe/flow/managers/listener/python_process_plugin.py [hunk 4]
```diff
@@ -481,14 +482,28 @@ def handle_create_object(
 
         :raises RuntimeError: if user has no permission to create the object.
         """
+
+        @retry(
+            max_retries=10,
```
Comment (romunov) [inline resolwe/flow/managers/listener/python_process_plugin.py:487]: Retries are hard coded here. Would it make sense to fetch this information from `resolwe.flow.models.base MAX_SLUG_RETRIES`?

**Review by romunov — APPROVED** [review]
Check spelling of colision in commit message, too.

File: docs/CHANGELOG.rst (hunk 1 above)
Comment (gregorjerse) [reply docs/CHANGELOG.rst:22]: Fixed.

File: resolwe/flow/models/base.py (hunk 2 above)
Comment (gregorjerse) [reply resolwe/flow/models/base.py:18]: Fixed.

File: resolwe/flow/models/base.py (hunk 3 above)
Comment (gregorjerse) [reply resolwe/flow/models/base.py:113]: Sure, that will add some information.

**Review by romunov — APPROVED** [review]

**Review by dblenkus — APPROVED** [review]

**Pull request merged**